Remotion/Core/Development.Analyzers.IntegrationTests/RMCORE0001_SafeContextAnalyzerTests.cs

```diff
@@ -0,0 +1,236 @@
+
+      var diagnosticForEach = Verifier.Diagnostic(Analyzer.AlternativeDescriptor)
+          .WithSpan(14, 5, 14, 78)
+          .WithMessage(
+              "'System.Threading.Tasks.Parallel.ForEach' should not be used with SafeContext by itself, use 'provider.OpenSafeContextBoundary()' to create a safe context boundary.");
+
+      var diagnosticForEachAsync = Verifier.Diagnostic(Analyzer.AlternativeDescriptor)
+          .WithSpan(16, 5, 20, 6)
+          .WithMessage(
+              "'System.Threading.Tasks.Parallel.ForEachAsync' should not be used with SafeContext by itself, use 'provider.OpenSafeContextBoundary()' to create a safe context boundary.");
+
+      var diagnosticInvoke = Verifier.Diagnostic(Analyzer.AlternativeDescriptor)
+          .WithSpan(22, 5, 22, 46)
+          .WithMessage(
+              "'System.Threading.Tasks.Parallel.Invoke' should not be used with SafeContext by itself, use 'provider.OpenSafeContextBoundary()' to create a safe context boundary.");
+
+      await Verifier.VerifyAnalyzerAsync(input, true,diagnosticFor, diagnosticForEach, diagnosticForEachAsync, diagnosticInvoke);
+    }
+
+    [Test]
+    public async Task SafeContext_NewThread_Diagnostic ()
+    {
+      const string input = @"
+using System.Threading;
+using Remotion.Context;
+public class A
+{
+  public void Test ()
+  { 
+    SafeContext.Instance.SetData(""a"", 3);
+    new Thread(() => SafeContext.Instance.GetData(""a""));
+  }
+}
+";
+      var diagnosticFunc = Verifier.Diagnostic(Analyzer.AlternativeDescriptor)
+          .WithSpan(9, 5, 9, 56)
+          .WithMessage(
+              "'new System.Threading.Thread' should not be used with SafeContext, use 'Remotion.Context.SafeContext.Thread.New' instead.");
+
+      await Verifier.VerifyAnalyzerAsync(input, true,diagnosticFunc);
+    }
+
+    [Test]
+    public async Task SafeContext_NewThreadingTimer_Diagnostic ()
+    {
+      const string input = @"
+using System.Threading;
+using Remotion.Context;
+public class A
+{
+  public void Test ()
+  { 
+    SafeContext.Instance.SetData(""a"", 3);
+    var timer = new Timer(new TimerCallback(_ => { }));
+  }
+}
+";
+      var diagnosticFunc = Verifier.Diagnostic(Analyzer.AlternativeDescriptor)
+          .WithSpan(9, 17, 9, 55)
+          .WithMessage(
+              "'new System.Threading.Timer' should not be used with SafeContext by itself, use 'provider.OpenSafeContextBoundary()' to create a safe context boundary.");
+
+      await Verifier.VerifyAnalyzerAsync(input, true,diagnosticFunc);
+    }
+
+    [Test]
+    public async Task SafeContext_NewTimersTimer_Diagnostic ()
+    {
+      const string input = @"
+using System.Timers;
+using Remotion.Context;
+public class A
+{
+  public void Test ()
+  { 
+    SafeContext.Instance.SetData(""a"", 3);
+    var timer = new Timer(new TimerCallback(_ => { }));
```

Comment:
Does that compile? Because `System.Timers.Timer` does not have a callback constructor. So in this case we want to check for the elapsed event and suggest `SafeContext.Timers.AddElapsedEventHandler`